### PR TITLE
Adding CE Loss logging to MNIST example

### PR DESCRIPTION
**Pull request by mortzur**
Summary:

* Adding runBatchReportLoss which allows a direct calculation of CrossEntropy loss from Softmax node, or alternatively extracts it from CrossEntropyLossNode if such a node is provided.
* Modify MNIST example to use the new function.

Testing:

Using verbose mode I got this loss graph -
Screen Shot 2019-07-29 at 11 39 53 AM

### CODE REVIEW

File: lib/ExecutionEngine/ExecutionEngine.cpp [hunk 1]
```diff
@@ -234,6 +234,65 @@ void glow::runBatch(ExecutionEngine &EE, PlaceholderBindings &bindings,
...
+      // Pick up one slice from the input tensors, and load it into the
+      // corresponding network Placeholder.
+      backingTensor->copyConsecutiveSlices(inputs[i], slc);
+    }
+
+    // Run the network.
+    EE.run(bindings);
+    sampleCounter += batchSize;
+
+    float ce_loss = 0;
```
Comment (putivsky) [inline lib/ExecutionEngine/ExecutionEngine.cpp]: Instead of writing new method with duplicate code, you could modify the old method providing a callback method as an input and execute callback on every iteration:

```
void glow::runBatchWithCallback(ExecutionEngine &EE,
                              PlaceholderBindings &bindings, size_t iterations,
                              size_t &sampleCounter,
                              llvm::ArrayRef<Placeholder *> ph,
                              llvm::ArrayRef<Tensor *> inputs,
                              Placeholder *resultPH, 
                             std::function<void()> &&callback == nullptr) {
   loop:
      
     if (callback) {
       callback();
     }

}

```

File: include/glow/ExecutionEngine/ExecutionEngine.h [hunk 2]
```diff
@@ -164,6 +164,30 @@ void runBatch(ExecutionEngine &EE, PlaceholderBindings &bindings,
               llvm::ArrayRef<Placeholder *> ph,
               llvm::ArrayRef<Tensor *> inputs);
 
+/// Runs \p iterations iterations of the compiled function. The method updates a
```
[the remark this answers is not in the document]
Comment (mortzur) [reply include/glow/ExecutionEngine/ExecutionEngine.h]: actually the first one is the name of the argument with will have italic/bold format on the doc..

**Comment by nadavrot** [comment]
@mortzur What is the status of this PR?
